### ISSUE TITLE
Driver exits and does not restart after a syntax error

### ISSUE THREAD

**Issue by smacker** [bblfsh/php-driver#31]
Bblfshd: v2.11.7
Driver: v2.7.0

Logs from the server:
```
FATAL <file>: syntax error: Syntax error, unexpected ')' on line 4
FATAL <file>: driver failure: unexpected end of JSON input
FATAL <file>: driver failure: native driver is not running
```

I can reproduce it locally:
```
root@a743b6d87529:/# bblfshctl parse fail.phpt
Status: Fatal
Elapsed: 1.2380209s
Errors:
	- syntax error: Syntax error, unexpected ')' on line 4


root@a743b6d87529:/# bblfshctl parse fail.phpt
Status: Fatal
Elapsed: 4.0194ms
Errors:
	- syntax error: Syntax error, unexpected ')' on line 4


root@a743b6d87529:/# bblfshctl parse fail.phpt
Status: Fatal
Elapsed: 3.6043ms
Errors:
	- syntax error: Syntax error, unexpected ')' on line 4


root@a743b6d87529:/# bblfshctl parse fail.phpt
Status: Fatal
Elapsed: 2.2709ms
Errors:
	- syntax error: Syntax error, unexpected ')' on line 4


root@a743b6d87529:/# bblfshctl parse fail.php
Status: Fatal
Elapsed: 1.8465598s
Errors:
	- driver failure: unexpected end of JSON input


root@a743b6d87529:/# bblfshctl parse fail.phpt
Status: Fatal
Elapsed: 1.4034ms
Errors:
	- driver failure: native driver is not running
```

Any request to PHP driver after that (with correct files too) end up with the same error: `driver failure: native driver is not running`

I'll send files in private

**Comment by juanjux** [comment]
I'm working on this. Currently:

- Native driver without the Go (annotation + SDK) layer or bblfshd: works.
- Go+native driver running on its own docker without bblfshd: crash.
- bblfshd calling the driver: crash.

I'll now check the Go driver without bblfshd and, if it crash (and I'm pretty sure it'll since no other part use JSON and the error is JSON related) I'll continue debugging from there.

**Comment by bzz** [comment]
This should have been fixed by now in https://github.com/bblfsh/sdk/pull/381 and needs to be tested on bblfshd v2.11.9 once it's out.

**Comment by dennwc** [comment]
@smacker Can you please send the file?

**Comment by smacker** [comment]
done in private

**Comment by dennwc** [comment]
The crash will be fixed by #41 and the root cause is #42.
